query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala

```diff
@@ -100,9 +100,13 @@ class TopBottomKRowAggregator(k: Int, bottomK: Boolean) extends RowAggregator {
    */
   private def createBuilder(rangeParams: RangeParams, currentTime: Long): RecordBuilder= {
     val builder = SerializedRangeVector.newBuilder();
-    for (t <- rangeParams.startSecs to (currentTime - rangeParams.stepSecs) by rangeParams.stepSecs) {
+    val it = Iterator.from(0, rangeParams.stepSecs.toInt)
+      .takeWhile(_ <= (currentTime - rangeParams.stepSecs) - rangeParams.startSecs).map { i =>
+      val t = i + rangeParams.startSecs
       addRecordToBuilder(builder, t * 1000, Double.NaN)
     }
+    // when step == 0 we don't need to add to builder
+    if (rangeParams.startSecs != rangeParams.endSecs) it.toList
```

Comment:
potential infinite loop here, can you add `step == 0` condition also

Reply:
Not needed here